internal/provider/base.go

```diff
@@ -25,6 +25,8 @@ type Provider interface {
 	CleanPlatformValuesFromDBUpdates(updates map[string]interface{}) error
 	// SetPlatformUsages uses the usageApi to update platform specific usages
 	SetPlatformUsages(platformParams *models.Platform, usages map[string]models.Usage, usageApi usage.API) error
+	// AreHostsSupported checks if the provider supports the hosts from values discovered in the inventory data
+	AreHostsSupported(hostInventories []*models.Inventory) (bool, error)
```

Comment:
1. Let's have two functions. One for single host validation and one for cluster hosts validations.
I think a validation for a single host could be easier to manipulate and run in a loop if necessary instead of just supporting arrays.
2. I think the wrapping models.Host for the Inventory is better since we can add more business logic metadata to the host without it being detected by the agent and as such sent as part of the Inventory model.

Reply:
Should be merged within the validator https://github.com/openshift/assisted-service/blob/9d64b25d5fcd2c90c360cab28b15b32288124c7e/internal/host/validator.go#L437-L454

Reply:
Done,

Regarding the isValidPlatform function I prefer to do it on a separate PR, so I opened `OCPRHV-622` to track it